test/e2e/route_advertisements.go

```diff
@@ -1161,11 +1198,154 @@ var _ = ginkgo.DescribeTableSubtree("BGP: isolation between advertised networks"
+				}),
+
+			ginkgo.Entry("[ETP=LOCAL]UDN pod to the same node nodeport service in different UDN network should not work",
+				func(ipFamily utilnet.IPFamily) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					clientPod := podNetB
+					node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), podsNetA[2].Spec.NodeName, metav1.GetOptions{})
+					framework.ExpectNoError(err)
+					nodeIPv4, nodeIPv6 := getNodeAddresses(node)
+					nodeIP := nodeIPv4
+					if ipFamily == utilnet.IPv6 {
+						nodeIP = nodeIPv6
+					}
+					nodePortA := svcNodePortETPLocalNetA.Spec.Ports[0].NodePort
+					out := curlConnectionTimeoutCode
+					errBool := true
+					if ipFamily == utilnet.IPv6 && cudnATemplate.Spec.Network.Topology == udnv1.NetworkTopologyLayer2 {
+						// For Layer2 networks, we have these flows we add on breth0:
+						// cookie=0xdeff105, duration=173.245s, table=1, n_packets=0, n_bytes=0, idle_age=173, priority=14,icmp6,icmp_type=134 actions=FLOOD
+						// cookie=0xdeff105, duration=173.245s, table=1, n_packets=8, n_bytes=640, idle_age=4, priority=14,icmp6,icmp_type=136 actions=FLOOD
+						// which floods the Router Advertisement (RA, type 134) and Neighbor Advertisement (NA, type 136)
+						// Given on Layer2 the GR has the SNATs for both masqueradeIPs this works perfectly well and
+						// the networks are able to NDP for the masqueradeIPs for the other networks.
+						// This doesn't work on Layer3 networks since masqueradeIP SNATs are present on the ovn-cluster-router in that case.
+						// See the tcpdump on the issue: https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5410 for more details.
+						out = ""
+						errBool = false
+					}
+					//FIXME https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5531
+					//For LGW mode, nodePort service with ETP=local is not working for UDN networks
+					if IsGatewayModeLocal(f.ClientSet) && cudnATemplate.Spec.Network.Topology == udnv1.NetworkTopologyLayer3 {
+						out = "56"
+						errBool = true
+					}
+					return clientPod.Name, clientPod.Namespace, net.JoinHostPort(nodeIP, fmt.Sprint(nodePortA)), out, errBool
+				}),
+			ginkgo.Entry("[ETP=LOCAL]UDN pod to a different node nodeport service in different UDN network should work",
+				func(ipFamily utilnet.IPFamily) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					clientPod := podNetB
+					node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), podsNetA[0].Spec.NodeName, metav1.GetOptions{})
+					framework.ExpectNoError(err)
+					nodeIPv4, nodeIPv6 := getNodeAddresses(node)
+					nodeIP := nodeIPv4
+					if ipFamily == utilnet.IPv6 {
+						nodeIP = nodeIPv6
+					}
+					nodePortA := svcNodePortETPLocalNetA.Spec.Ports[0].NodePort
+					out := ""
+					errBool := false
+					//FIXME https://github.com/ovn-kubernetes/ovn-kubernetes/issues/5531
```

Comment:
super nit: all comments usually have a space after the backstrokes.... surprising the IDE is not catching this.